batched/dense/impl/KokkosBatched_Gemm_Common_Impl.hpp

```diff
@@ -0,0 +1,79 @@
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 4.0
+//       Copyright (2022) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+#ifndef KOKKOSBATCHED_GEMM_COMMON_IMPL_HPP
+#define KOKKOSBATCHED_GEMM_COMMON_IMPL_HPP
+
+#include "KokkosBlas_util.hpp"
+#include "KokkosBatched_Util.hpp"
+
+namespace KokkosBatched {
+namespace Impl {
+template <typename ArgTransA, typename ArgTransB, typename AViewType, typename BViewType, typename CViewType>
+KOKKOS_INLINE_FUNCTION static int checkGemmInput([[maybe_unused]] const AViewType &A,
+                                                 [[maybe_unused]] const BViewType &B,
+                                                 [[maybe_unused]] const CViewType &C) {
+  static_assert(Kokkos::is_view_v<AViewType>, "KokkosBatched::gemm: AViewType is not a Kokkos::View.");
+  static_assert(Kokkos::is_view_v<BViewType>, "KokkosBatched::gemm: BViewType is not a Kokkos::View.");
+  static_assert(Kokkos::is_view_v<CViewType>, "KokkosBatched::gemm: CViewType is not a Kokkos::View.");
+
+  static_assert(AViewType::rank <= 2, "KokkosBatched::gemm: AViewType must have rank 0, 1 or 2.");
```

Comment:
Please use `rank()` instead, see View documentation in Kokkos Core API.

Reply:
Yes, I have updated accordingly